docker/Dockerfile

```diff
@@ -0,0 +1,88 @@
+FROM ubuntu:22.04
+MAINTAINER Weina Ji <weina.ji@epfl.ch>
+
+# The default shell for the RUN instruction is ["/bin/sh", "-c"].
+# Using SHELL instruction to change default shell for subsequent RUN instructions
+SHELL ["/bin/bash", "-c"]
+
+ARG LIBSONATA_TAG=master
+ARG LIBSONATAREPORT_TAG=master
+ARG NEURON_TAG=master
```

Comment:
@pramodk @heerener 
I've added 3 args for the tag version in each package. The default value is `master` not the latest tag because sorting tags in different repo can give surprising results.
In GitLab workflow, we will build with the right tag versions used by the spack recipe.